src/main/java/io/confluent/idesidecar/restapi/credentials/TLSConfig.java

```diff
@@ -0,0 +1,304 @@
+package io.confluent.idesidecar.restapi.credentials;
+
+import com.fasterxml.jackson.annotation.JsonEnumDefaultValue;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import io.confluent.idesidecar.restapi.exceptions.Failure;
+import io.confluent.idesidecar.restapi.exceptions.Failure.Error;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Null;
+import jakarta.validation.constraints.Size;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+@Schema(description = "SSL configuration")
+public record TLSConfig(
+    @Schema(description = "Whether to verify the server's SSL certificate. If set to false, "
+        + "the server's SSL certificate will not be verified. Defaults to true.")
+    @JsonProperty(value = "verify_ssl_certificates")
+    @Null
+    Boolean verifySslCertificates,
+    @Schema(description =
+        "The trust store configuration for authenticating the server's certificate.")
+    @JsonProperty(value = "truststore")
+    @NotNull
+    TrustStore truststore,
+    @Schema(description = "The key store configuration for "
+        + "authenticating the client to the server.",
+        nullable = true)
```

Comment:
Should we explain that this need be provided only when using mTLS? Maybe something like:
```suggestion
    @Schema(description = "The key store configuration that will identify and authenticate the client to the server, required for mutual TLS (mTLS)",
        nullable = true)
```